Vendor cadvisor v0.38.4

bobbypage opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**
/kind bug
<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature
/kind design

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

**What this PR does / why we need it**:

Vendor in cadvisor v0.38.4

For each kubernetes release, we need to cut a new cadvisor release and vendor it back into k/k. For 1.20, we'll be vendoring cadvisor v0.38.4.


**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #96287

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```
```

```
hack/pin-dependency.sh github.com/google/cadvisor v0.38.4
hack/update-vendor.sh


$ hack/lint-dependencies.sh
These modules are pinned to versions different than the minimal preferred version.
That means that without replace directives, a different version would be selected,
which breaks consumers of our published modules.
1. Use hack/pin-dependency.sh to switch to the preferred version for each module
2. Run hack/update-vendor.sh to rebuild the vendor directory
3. Run hack/lint-dependencies.sh to verify no additional changes are required

cloud.google.com/go
    pinned:    v0.51.0
    preferred: v0.54.0
    hack/pin-dependency.sh cloud.google.com/go v0.54.0
github.com/aws/aws-sdk-go
    pinned:    v1.35.5
    preferred: v1.35.24
    hack/pin-dependency.sh github.com/aws/aws-sdk-go v1.35.24
github.com/blang/semver
    pinned:    v3.5.0+incompatible
    preferred: v3.5.1+incompatible
    hack/pin-dependency.sh github.com/blang/semver v3.5.1+incompatible
github.com/containerd/containerd
    pinned:    v1.3.3
    preferred: v1.4.1
    hack/pin-dependency.sh github.com/containerd/containerd v1.4.1
github.com/containerd/ttrpc
    pinned:    v1.0.0
    preferred: v1.0.2
    hack/pin-dependency.sh github.com/containerd/ttrpc v1.0.2
github.com/containerd/typeurl
    pinned:    v1.0.0
    preferred: v1.0.1
    hack/pin-dependency.sh github.com/containerd/typeurl v1.0.1
github.com/docker/docker
    pinned:    v1.4.2-0.20200309214505-aa6a9891b09c
    preferred: v17.12.0-ce-rc1.0.20200916142827-bd33bbf0497b+incompatible
    hack/pin-dependency.sh github.com/docker/docker v17.12.0-ce-rc1.0.20200916142827-bd33bbf0497b+incompatible
github.com/golang/protobuf
    pinned:    v1.4.2
    preferred: v1.4.3
    hack/pin-dependency.sh github.com/golang/protobuf v1.4.3
github.com/google/go-cmp
    pinned:    v0.4.0
    preferred: v0.5.2
    hack/pin-dependency.sh github.com/google/go-cmp v0.5.2
github.com/google/uuid
    pinned:    v1.1.1
    preferred: v1.1.2
    hack/pin-dependency.sh github.com/google/uuid v1.1.2
github.com/gorilla/mux
    pinned:    v1.7.3
    preferred: v1.8.0
    hack/pin-dependency.sh github.com/gorilla/mux v1.8.0
github.com/karrick/godirwalk
    pinned:    v1.7.5
    preferred: v1.16.1
    hack/pin-dependency.sh github.com/karrick/godirwalk v1.16.1
github.com/opencontainers/runc
    pinned:    v1.0.0-rc91.0.20200707015106-819fcc687efb
    preferred: v1.0.0-rc92
    hack/pin-dependency.sh github.com/opencontainers/runc v1.0.0-rc92
github.com/opencontainers/runtime-spec
    pinned:    v1.0.3-0.20200520003142-237cc4f519e2
    preferred: v1.0.3-0.20200728170252-4d89ac9fbff6
    hack/pin-dependency.sh github.com/opencontainers/runtime-spec v1.0.3-0.20200728170252-4d89ac9fbff6
github.com/prometheus/procfs
    pinned:    v0.1.3
    preferred: v0.2.0
    hack/pin-dependency.sh github.com/prometheus/procfs v0.2.0
github.com/stretchr/testify
    pinned:    v1.4.0
    preferred: v1.6.1
    hack/pin-dependency.sh github.com/stretchr/testify v1.6.1
github.com/vishvananda/netns
    pinned:    v0.0.0-20200520041808-52d707b772fe
    preferred: v0.0.0-20200728191858-db3c7e526aae
    hack/pin-dependency.sh github.com/vishvananda/netns v0.0.0-20200728191858-db3c7e526aae
golang.org/x/text
    pinned:    v0.3.3
    preferred: v0.3.4
    hack/pin-dependency.sh golang.org/x/text v0.3.4
golang.org/x/time
    pinned:    v0.0.0-20191024005414-555d28b269f0
    preferred: v0.0.0-20200630173020-3af7569d3a1e
    hack/pin-dependency.sh golang.org/x/time v0.0.0-20200630173020-3af7569d3a1e
golang.org/x/xerrors
    pinned:    v0.0.0-20191204190536-9bdfabe68543
    preferred: v0.0.0-20200804184101-5ec99f83aff1
    hack/pin-dependency.sh golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1
google.golang.org/genproto
    pinned:    v0.0.0-20200526211855-cb27e3aa2013
    preferred: v0.0.0-20201110150050-8816d57aaa9a
    hack/pin-dependency.sh google.golang.org/genproto v0.0.0-20201110150050-8816d57aaa9a
google.golang.org/grpc
    pinned:    v1.27.0
    preferred: v1.27.1
    hack/pin-dependency.sh google.golang.org/grpc v1.27.1
google.golang.org/protobuf
    pinned:    v1.24.0
    preferred: v1.25.0
    hack/pin-dependency.sh google.golang.org/protobuf v1.25.0
k8s.io/utils
    pinned:    v0.0.0-20201104234853-8146046b121e
    preferred: v0.0.0-20201110183641-67b214c5f920
    hack/pin-dependency.sh k8s.io/utils v0.0.0-20201110183641-67b214c5f920

Use the given commands to remove pinned module versions that aren't actually used:
GO111MODULE=on go mod edit -dropreplace github.com/go-ini/ini
All pinned versions of checked dependencies match their preferred version.

# pin all of the above
$ hack/update-vendor.sh



$ hack/lint-dependencies.sh
These modules are pinned to versions different than the minimal preferred version.
That means that without replace directives, a different version would be selected,
which breaks consumers of our published modules.
1. Use hack/pin-dependency.sh to switch to the preferred version for each module
2. Run hack/update-vendor.sh to rebuild the vendor directory
3. Run hack/lint-dependencies.sh to verify no additional changes are required

cloud.google.com/go/bigquery
    pinned:    v1.0.1
    preferred: v1.4.0
    hack/pin-dependency.sh cloud.google.com/go/bigquery v1.4.0
cloud.google.com/go/datastore
    pinned:    v1.0.0
    preferred: v1.1.0
    hack/pin-dependency.sh cloud.google.com/go/datastore v1.1.0
cloud.google.com/go/pubsub
    pinned:    v1.0.1
    preferred: v1.2.0
    hack/pin-dependency.sh cloud.google.com/go/pubsub v1.2.0
cloud.google.com/go/storage
    pinned:    v1.0.0
    preferred: v1.6.0
    hack/pin-dependency.sh cloud.google.com/go/storage v1.6.0
github.com/checkpoint-restore/go-criu/v4
    pinned:    v4.0.2
    preferred: v4.1.0
    hack/pin-dependency.sh github.com/checkpoint-restore/go-criu/v4 v4.1.0
github.com/golang/mock
    pinned:    v1.3.1
    preferred: v1.4.1
    hack/pin-dependency.sh github.com/golang/mock v1.4.1
github.com/google/pprof
    pinned:    v0.0.0-20191218002539-d4f498aebedc
    preferred: v0.0.0-20200229191704-1ebb73c60ed3
    hack/pin-dependency.sh github.com/google/pprof v0.0.0-20200229191704-1ebb73c60ed3
github.com/opencontainers/selinux
    pinned:    v1.5.2
    preferred: v1.6.0
    hack/pin-dependency.sh github.com/opencontainers/selinux v1.6.0
go.opencensus.io
    pinned:    v0.22.2
    preferred: v0.22.3
    hack/pin-dependency.sh go.opencensus.io v0.22.3
golang.org/x/lint
    pinned:    v0.0.0-20191125180803-fdd1cda4f05f
    preferred: v0.0.0-20200302205851-738671d3881b
    hack/pin-dependency.sh golang.org/x/lint v0.0.0-20200302205851-738671d3881b
golang.org/x/oauth2
    pinned:    v0.0.0-20191202225959-858c2ad4c8b6
    preferred: v0.0.0-20200107190931-bf48bf16ab8d
    hack/pin-dependency.sh golang.org/x/oauth2 v0.0.0-20200107190931-bf48bf16ab8d
google.golang.org/api
    pinned:    v0.15.1
    preferred: v0.20.0
    hack/pin-dependency.sh google.golang.org/api v0.20.0
google.golang.org/protobuf
    pinned:    v1.24.0
    preferred: v1.25.0
    hack/pin-dependency.sh google.golang.org/protobuf v1.25.0
honnef.co/go/tools
    pinned:    v0.0.1-2019.2.3
    preferred: v0.0.1-2020.1.3
    hack/pin-dependency.sh honnef.co/go/tools v0.0.1-2020.1.3

Use the given commands to remove pinned module versions that aren't actually used:
GO111MODULE=on go mod edit -dropreplace github.com/docker/go-events
GO111MODULE=on go mod edit -dropreplace github.com/go-ini/ini
GO111MODULE=on go mod edit -dropreplace github.com/gogo/googleapis
All pinned versions of checked dependencies match their preferred version.

# pin all of the above
$ hack/update-vendor.sh

$ hack/lint-dependencies.sh
These modules are pinned to versions different than the minimal preferred version.
That means that without replace directives, a different version would be selected,
which breaks consumers of our published modules.
1. Use hack/pin-dependency.sh to switch to the preferred version for each module
2. Run hack/update-vendor.sh to rebuild the vendor directory
3. Run hack/lint-dependencies.sh to verify no additional changes are required

github.com/golang/groupcache
    pinned:    v0.0.0-20191227052852-215e87163ea7
    preferred: v0.0.0-20200121045136-8c9f03a8e57e
    hack/pin-dependency.sh github.com/golang/groupcache v0.0.0-20200121045136-8c9f03a8e57e
golang.org/x/exp
    pinned:    v0.0.0-20191227195350-da58074b4299
    preferred: v0.0.0-20200224162631-6cc2880d07d6
    hack/pin-dependency.sh golang.org/x/exp v0.0.0-20200224162631-6cc2880d07d6
google.golang.org/protobuf
    pinned:    v1.24.0
    preferred: v1.25.0
    hack/pin-dependency.sh google.golang.org/protobuf v1.25.0

Use the given commands to remove pinned module versions that aren't actually used:
GO111MODULE=on go mod edit -dropreplace github.com/docker/go-events
GO111MODULE=on go mod edit -dropreplace github.com/go-ini/ini
GO111MODULE=on go mod edit -dropreplace github.com/gogo/googleapis
All pinned versions of checked dependencies match their preferred version.

# pin all of the above
$ hack/update-vendor.sh

$ hack/lint-dependencies.sh
These modules are pinned to versions different than the minimal preferred version.
That means that without replace directives, a different version would be selected,
which breaks consumers of our published modules.
1. Use hack/pin-dependency.sh to switch to the preferred version for each module
2. Run hack/update-vendor.sh to rebuild the vendor directory
3. Run hack/lint-dependencies.sh to verify no additional changes are required

github.com/go-gl/glfw/v3.3/glfw
    pinned:    v0.0.0-20191125211704-12ad95a8df72
    preferred: v0.0.0-20200222043503-6f7a984d4dc4
    hack/pin-dependency.sh github.com/go-gl/glfw/v3.3/glfw v0.0.0-20200222043503-6f7a984d4dc4
google.golang.org/protobuf
    pinned:    v1.24.0
    preferred: v1.25.0
    hack/pin-dependency.sh google.golang.org/protobuf v1.25.0

Use the given commands to remove pinned module versions that aren't actually used:
GO111MODULE=on go mod edit -dropreplace github.com/docker/go-events
GO111MODULE=on go mod edit -dropreplace github.com/go-ini/ini
GO111MODULE=on go mod edit -dropreplace github.com/gogo/googleapis
All pinned versions of checked dependencies match their preferred version.

# pin all of the above
$ hack/update-vendor.sh

$ hack/lint-dependencies.sh
These modules are pinned to versions different than the minimal preferred version.
That means that without replace directives, a different version would be selected,
which breaks consumers of our published modules.
1. Use hack/pin-dependency.sh to switch to the preferred version for each module
2. Run hack/update-vendor.sh to rebuild the vendor directory
3. Run hack/lint-dependencies.sh to verify no additional changes are required

google.golang.org/protobuf
    pinned:    v1.24.0
    preferred: v1.25.0
    hack/pin-dependency.sh google.golang.org/protobuf v1.25.0

Use the given commands to remove pinned module versions that aren't actually used:
GO111MODULE=on go mod edit -dropreplace github.com/docker/go-events
GO111MODULE=on go mod edit -dropreplace github.com/go-ini/ini
GO111MODULE=on go mod edit -dropreplace github.com/gogo/googleapis
All pinned versions of checked dependencies match their preferred version.

# pin protobuf
hack/update-vendor.sh

$ hack/lint-dependencies.sh

Use the given commands to remove pinned module versions that aren't actually used:
GO111MODULE=on go mod edit -dropreplace github.com/docker/go-events
GO111MODULE=on go mod edit -dropreplace github.com/go-ini/ini
GO111MODULE=on go mod edit -dropreplace github.com/gogo/googleapis
All pinned versions of checked dependencies match their preferred version.

$ hack/update-vendor.sh

$ hack/lint-dependencies.sh
All pinned versions of checked dependencies match their preferred version.

$ hack/verify-vendor.sh
+++ [1113 19:51:08] logfile at /tmp/update-vendor.f32t/update-vendor.log
+++ [1113 19:51:08] go.mod: update staging references
+++ [1113 19:51:19] go.mod: propagate to staging modules
+++ [1113 19:51:21] go.mod: sorting staging modules
+++ [1113 19:51:45] go.mod: tidying
+++ [1113 19:52:08] go.mod: adding generated comments
+++ [1113 19:52:09] vendor: running 'go mod vendor'
+++ [1113 19:52:10] vendor: updating BUILD files
+++ [1113 19:52:17] vendor: updating vendor/LICENSES
+++ [1113 19:52:26] vendor: creating OWNERS file
+++ [1113 19:52:26] NOTE: don't forget to handle vendor/* files that were added or removed
All pinned versions of checked dependencies match their preferred version.
Vendor Verified.
```

bobbypage commented:
/cc @dashpole @dims @SergeyKanzhelev

dashpole commented:
This seems like it may be blocked by https://github.com/kubernetes/kubernetes/issues/93320

fedebongio commented:
/remove-sig api-machinery

bobbypage commented:
/sig node

bobbypage commented:
Ran into issues due to GRPC update to 1.33.2 in cadvisor from https://github.com/google/cadvisor/pull/2716 due to a change in grpc which broke etcd (https://github.com/kubernetes/kubernetes/issues/93320#issuecomment-685237947).

Rolled back grpc to 1.27.1 in cadvisor and cut a new release, v0.38.1

bobbypage commented:
We hit another issue due to cadvisor importing a staging repo causing a recursion: https://github.com/google/cadvisor/pull/2726

Thanks @dims for the fix there. I cut another patch release v0.38.2 with the fix. Let's try to vendor again :)

bobbypage commented:
/retest

bobbypage commented:
Looking better now, the tests are alot more green!

bobbypage commented:
/kind bug
/milestone v1.20

bobbypage commented:
/retest

bobbypage commented:
/retest

dims commented:
/approve
/lgtm

/assign @liggitt  (for root approval)

bobbypage commented:
/retest

bobbypage commented:
/retest

JornShen commented:
@bobbypage, need to update the same dependency of cadvisor to k8s v1.19 to fix the problem https://github.com/kubernetes/kubernetes/issues/94281.  :)

bobbypage commented:
Thanks @JornShen, we might need to do a cherrypick for that issue if necessary.

bobbypage commented:
Almost there, but one of the the e2e tests is failing `pull-kubernetes-bazel-test`. Specifically, `//test/conformance/conformance_test:run_1_of_2` is failing.

Chatting with @dims offline, failure is due to the vendor update of `gopkg.in/yaml.v2 v2.2.8` to `gopkg.in/yaml.v2 v2.3.0`. Apparently v2.3.0 has a [breaking change](https://github.com/go-yaml/yaml/pull/571) that changed the default max YAML line length from 80 characters to unlimited characters. 

This is an issue for us because [Conformance test](https://github.com/kubernetes/kubernetes/blob/5e44d8ebafc866e132530fa1328872561ba5f6da/test/conformance/walk.go) generates some YAML config and does a git diff from what is checked into the repo.  Due to this change the conformance tests are failing since there is a diff in the output yaml due to the line length change of this library.  

More discussion about this issue in here as well: https://github.com/kubernetes/kubernetes/pull/95571#issuecomment-708796327

Next steps sounds like one of the following options: 

1. Attempt to revert back to v2.2.8 of the library 
  * (need to check if this will be possible or one of the transitive dependencies needs >= v2.3.0
2. See if we can somehow force the library to use 80 character limit as before
3. Regen the Conformance YAML under v2.3.0 of the library

bobbypage commented:
Option #1 sounds like it should be ok, I don't see any usages of `gopkg.in/yaml.v2` with version >= v2.3.0:

```
$ rg --type-add 'mod:*.mod' -tmod "gopkg.in/yaml.v2"
go.mod
111:    gopkg.in/yaml.v2 v2.2.8
487:    gopkg.in/yaml.v2 => gopkg.in/yaml.v2 v2.2.8

vendor/gopkg.in/yaml.v2/go.mod
1:module "gopkg.in/yaml.v2"

vendor/sigs.k8s.io/yaml/go.mod
7:      gopkg.in/yaml.v2 v2.2.8

vendor/github.com/Microsoft/hcsshim/go.mod
33:     gopkg.in/yaml.v2 v2.2.8 // indirect

vendor/github.com/go-openapi/swag/go.mod
9:      gopkg.in/yaml.v2 v2.2.2

vendor/github.com/go-openapi/spec/go.mod
14:     gopkg.in/yaml.v2 v2.2.2

vendor/github.com/onsi/gomega/go.mod
14:     gopkg.in/yaml.v2 v2.2.1

vendor/github.com/go-openapi/runtime/go.mod
15:     gopkg.in/yaml.v2 v2.2.2

vendor/github.com/go-openapi/validate/go.mod
17:     gopkg.in/yaml.v2 v2.2.4

vendor/github.com/go-openapi/loads/go.mod
8:      gopkg.in/yaml.v2 v2.2.4

vendor/github.com/spf13/viper/go.mod
39:     gopkg.in/yaml.v2 v2.2.4

vendor/github.com/gophercloud/gophercloud/go.mod
6:      gopkg.in/yaml.v2 v2.2.2

vendor/github.com/spf13/cobra/go.mod
11:     gopkg.in/yaml.v2 v2.2.8

staging/src/k8s.io/apiextensions-apiserver/go.mod
23:     gopkg.in/yaml.v2 v2.2.8

staging/src/k8s.io/cli-runtime/go.mod
21:     gopkg.in/yaml.v2 v2.2.8

staging/src/k8s.io/kubectl/go.mod
36:     gopkg.in/yaml.v2 v2.2.8

staging/src/k8s.io/mount-utils/go.mod
11:     gopkg.in/yaml.v2 v2.2.8 // indirect

staging/src/k8s.io/apimachinery/go.mod
34:     gopkg.in/yaml.v2 v2.2.8

staging/src/k8s.io/cri-api/go.mod
19:     gopkg.in/yaml.v2 v2.2.8 // indirect

staging/src/k8s.io/apiserver/go.mod
43:     gopkg.in/yaml.v2 v2.2.8
```

bobbypage commented:
option #2 doesn't sound possible -- there is a v3 version of the library but that also doesn't support line length (same behavior of unlimited length line as v2.3.0) ref: https://github.com/go-yaml/yaml/pull/455

bobbypage commented:
I think the reason that `gopkg.in/yaml.v2` needed to be updated from `v2.2.8` to `v2.3.0` is because https://github.com/google/cadvisor/pull/2716 upgraded `github.com/prometheus/common` from `v0.10.0` to `v0.15.0`. 

As part of that upgrade, looks like `github.com/prometheus/common@v0.15.0` requires `gopkg.in/yaml.v2@v2.3.0` ref: https://github.com/prometheus/common/pull/242

dims commented:
@bobbypage let's regen the conformance yaml :) was just trying to avoid some extra approvals. We'll ping people needed.

liggitt commented:
> @bobbypage let's regen the conformance yaml :)

no, the yaml bump would affect all serialized yaml output. see identical discussion in https://github.com/kubernetes/kubernetes/pull/95571#issuecomment-709315815

fedebongio commented:
/remove-sig api-machinery

bobbypage commented:
New cadvisor v0.38.3 has been cut with the `gopkg.in/yaml.v2` reverted back to `v2.2.8`. https://github.com/google/cadvisor/pull/2728 

Let's try again... :)

dashpole commented:
/lgtm

bobbypage commented:
/assign @dims @liggitt

aojea commented on staging/src/k8s.io/apimachinery/go.mod:
```diff
@@ -25,11 +25,12 @@ require (
-	google.golang.org/protobuf v1.24.0 // indirect
+	github.com/stretchr/testify v1.6.1
+	golang.org/x/net v0.0.0-20201110031124-69a78807bb2b
+	golang.org/x/sys v0.0.0-20201110211018-35f3e6cf4a65 // indirect
```
@caesarxuchao I think that this will unblock you

caesarxuchao replied:
Thanks.

dchen1107 commented:
/lgtm
/approve

liggitt commented:
The approach the go.uber.org libraries have taken to expressing their development tool dependencies is really unfortunate, because it passes those development-time dependencies on to their library consumers.

The majority of the newly vendored dependencies are because of the tools.go files in `go.uber.org/{atomic,multierr,zap}`

Setting up a test copy of those modules that deletes that tools.go file and revendoring drops all of the following:

* vendor/github.com/BurntSushi/toml
* vendor/go.uber.org/tools
* vendor/golang.org/x/lint
* vendor/honnef.co/go/tools

I'm not sure what to do with vendor/honnef.co/go/tools/LICENSE-THIRD-PARTY ... our license gathering script doesn't aggregate it, and I'm not sure whether it should or not.

liggitt commented on vendor/github.com/opencontainers/runtime-spec/specs-go/config.go:
```diff
@@ -90,7 +90,7 @@ type User struct {
 	GID uint32 `json:"gid" platform:"linux,solaris"`
 	// Umask is the umask for the init process.
-	Umask uint32 `json:"umask,omitempty" platform:"linux,solaris"`
+	Umask *uint32 `json:"umask,omitempty" platform:"linux,solaris"`
```
that's a fun change... swept and didn't see any uses that would newly NPE

liggitt reviewed (commented):
cadvisor more than doubled the number of dependencies as seen by `go list -m all` between v0.37.0 and v0.38.3:

```
cadvisor]$ git checkout v0.37.0 && go list -m all | wc -l
HEAD is now at 65fa5b44 Merge pull request #2613 from dashpole/changelog-v0.37.0
     112

cadvisor]$ git checkout v0.38.3 && go list -m all | wc -l
HEAD is now at e02cd932  Rollback gopkg.in/yaml.v2 to v2.2.8
     271
```

is that expected? the number of new deps coming in with this is way more than I expected

bobbypage commented:
Thanks for looking more closely at the vendored changes. I looked into the increase of deps and it seems it was caused by https://github.com/google/cadvisor/pull/2716 

```
$ git checkout f6f48d031d729c3b38629c63e5c6ebb1a6ca76fc && go list -m all | wc -l
275
```

Before it was at 137 deps

```
$ git checkout f6f48d031d729c3b38629c63e5c6ebb1a6ca76fc^1 && go list -m all | wc -l
Previous HEAD position was f6f48d03 Update dependencies
HEAD is now at bf8fe422 Delete exclude guest flag which causing problems on some configurations. (#2640)
137
```
The reason for the increase deps in https://github.com/google/cadvisor/pull/2716  was probably due to upgrade of GRPC, but we reverted that in https://github.com/google/cadvisor/pull/2724 and also the `gopkg.in/yaml.v2` https://github.com/google/cadvisor/pull/2728. The issue looks like somewhere along the way we forget to run `go mod tidy` 😮. 

I ran `go mod tidy` and we're back to reasonable amount of deps, 140:

```
$ go list -m all | wc -l
140
```

cadvisor should really add a tidy check to CI, I filed https://github.com/google/cadvisor/issues/2730 for us to do that. 

I cut another cadvisor release after running tidy, https://github.com/google/cadvisor/pull/2731, cadvisor `v0.38.4` and vendored it back, and updated the PR. PTAL again, thanks again for catching this.

bobbypage commented:
/retest

liggitt commented:
This looks far more reasonable, thanks for the quick turnaround.

https://github.com/kubernetes/kubernetes/pull/96549 should go in first, since it is a targeted dep update of x/net and x/sys to address the http/2 bug, then this will need a rebase.

/lgtm
/approve
/hold for https://github.com/kubernetes/kubernetes/pull/96549

liggitt commented:
/priority important-soon
/triage accepted

dims commented:
/milestone v1.20

bobbypage committed:
vendor: cadvisor v0.38.4

bobbypage commented:
I've rebased.

bobbypage commented:
/retest

liggitt commented:
/hold cancel

liggitt commented:
/lgtm

bobbypage commented:
/retest
